kernel/nativelib.ml

```diff
@@ -48,6 +48,9 @@ let () = at_exit (fun () ->
           Pp.(str "Native compile: failed to cleanup: " ++
               str(Printexc.to_string e) ++ fnl()))
 
+let delay_cleanup_file f =
+  at_exit (fun () -> if not (keep_debug_files()) && Sys.file_exists f then Sys.remove f)
```

Comment:
Could you use a `string list ref` and a single `at_exit` call, instead of implicitly relying on the closure registration? It would make the intent clearer. Also, you should probably extrude the flag check.